samples/apps/copilot-chat-app/webapi/Controllers/DocumentImportController.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.SemanticFunctions.Partitioning;
+using SemanticKernel.Service.Config;
+using SemanticKernel.Service.Model;
+using SemanticKernel.Service.Skills;
+
+namespace SemanticKernel.Service.Controllers;
+
+/// <summary>
+/// Controller for importing documents.
+/// </summary>
+[ApiController]
+public class DocumentImportController : ControllerBase
+{
+    /// <summary>
+    /// Supported file types for import.
+    /// </summary>
+    internal enum SupportedFileType
+    {
+        TXT,    // .txt
+        PDF,    // .pdf
+    };
+    private readonly IServiceProvider _serviceProvider;
+    private readonly IConfiguration _configuration;
+    private readonly ILogger<DocumentImportController> _logger;
+    private readonly PromptSettings _promptSettings;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DocumentImportController"/> class.
+    /// </summary>
+    public DocumentImportController(
+        IServiceProvider serviceProvider,
+        IConfiguration configuration,
```

Comment:
You may want to take a look at https://github.com/microsoft/semantic-kernel/pull/588/files#diff-c5575c7287ebabe44c1f95b66955d5f4dc626c415ad12067a08c0521f35fc3f9 for a slightly different pattern of bringing in controller configurations (See IOptions).